### PR TITLE
[PWA-2228] Shimmer loader for dynamic blocks

### PR DESCRIPTION
**Pull request by jeremycharron**
## Description

Add shimmer loader on dynamic blocks to avoid layout shift.
Shimmer loaders will only be rendered if minHeight is specified on the row containing a dynamic block
This gives flexibility to users to display a shimmer or not (let's say a dynamic block wouldn't render most of the time, we probably don't want to display a shimmer)

Following page showcase different combinations 
Use browser speed throttling if you can't see them.
https://jeremycharron-o4saqja-4txwfa4z2sor6.eu-4.magentosite.cloud/dynamic-block-shimmers

HTML code of CMS Page in PWA-2228 ticket

## Related Issue

Closes PWA-2228

### Verification Stakeholders

Project maintainers

#### Test scenario(s) for direct fix/feature

1. Add a row with a minHeight, then add a dynamic block inside. Verify that shimmer is rendered.
2. Add a row with a minHeight, then add 2 dynamic blocks inside. Verify that only one shimmer is displayed
3. Add a row with a minHeight, then add a dynamic block with conditions that cause the block not to render on front-end. Verify that shimmer still renders (this is expected), and white-space is removed when empty content is loaded.
4. Add a row, then add dynamic block inside. Verify that no shimmer is rendered (current Develop behavior)
5. Add a row with a minHeight, then add a column and add a dynamic block inside. Verify that shimmer is displayed

#### Test scenario(s) for any existing impacted features/areas

1. Verify that new and existing Jest tests pass
2. Verify that Cypress test passes

#### Test scenario(s) for any Magento Backend Supported Configurations

N/A

#### Is Browser/Device testing needed?

Yes, multiple browsers and devices.

#### Any ad-hoc/edge case scenarios that need to be considered?

Should be 100% compatible with recently added Personalized Sample data content (https://github.com/magento-commerce/venia-sample-data-modules-ee).

## Screenshots / Screen Captures (if appropriate)
Using page shared above, 

Develop without minHeight :
cls-before-2

jcharron/PWA-2228 with minHeight :
CLS

Only the 4th test gets pushed, which is normal.
cls-2

## Breaking Changes (if any)

No breaking change

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by jeremycharron** [comment]
run cypress

File: packages/pagebuilder/lib/ContentTypes/DynamicBlock/dynamicBlock.shimmer.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        paddingBottom,
+        paddingLeft
+    };
+
+    // Only render shimmer if min-height specified
+    return minHeight ? (
+        <div className={classes.parent} style={rootStyles}>
+            <Shimmer
+                aria-live="polite"
+                aria-busy="true"
```
Comment (mikhaelbois) [inline packages/pagebuilder/lib/ContentTypes/DynamicBlock/dynamicBlock.shimmer.js:56]: Aria labels should also be added to the main component to indicate the block has finished loading.
You can refer to Banner component for an example.

File: packages/pagebuilder/lib/ContentTypes/DynamicBlock/configAggregator.js [hunk 2]
```diff
@@ -10,7 +10,10 @@ export default node => {
         return {};
     }
 
+    const row = widgetBlock.closest('[data-content-type="row"]')?.firstChild;
```
[the remark this answers is not in the document]
Comment (jeremycharron) [reply packages/pagebuilder/lib/ContentTypes/DynamicBlock/configAggregator.js:13]: As discussed with Olena, minHeight is always going to be set on the row level.
If a user wants a shimmer on a dynamic block in a column, they'll have to add it in a row.
Do you think code on line 13 is a valid approach in that case?

[the remark this answers is not in the document]
Comment (mikhaelbois) [reply packages/pagebuilder/lib/ContentTypes/DynamicBlock/configAggregator.js:13]: Should we had a comment explaining it?

File: packages/pagebuilder/lib/ContentTypes/DynamicBlock/dynamicBlock.shimmer.js (hunk 1 above)
Comment (jeremycharron) [reply packages/pagebuilder/lib/ContentTypes/DynamicBlock/dynamicBlock.shimmer.js]: Moved aria labels to parent element.
Also created https://jira.corp.magento.com/browse/PWA-2620 to refactor `aria-busy` attributes as a global element.
cc @JustinConabreeAbsolunet

File: packages/pagebuilder/lib/ContentTypes/DynamicBlock/configAggregator.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremycharron) [reply packages/pagebuilder/lib/ContentTypes/DynamicBlock/configAggregator.js:14]: Added comment.

**Comment by jeremycharron** [comment]
Updated PR with improvements for dynamic blocks that are not being the first element of a row.

**Review by mikhaelbois — APPROVED** [review]
LGTM ✅

**Comment by pedrochiossi** [comment]
QA tests:

1. Dynamic block renders shimmer if inside a row with minHeight ✅
2. Two Dynamic blocks renders only one shimmer ✅
3. Unmatched dynamic blocks for customer segment renders shimmer and shows empty content with white space removed after successful load ✅
4. Dynamic block does not render shimmer if inside a row with minHeight unset ✅ 
5. Dynamic block renders shimmer if inside a column inside a row with minHeight ✅
6. Working on Firefox v97.0.1 ✅
7. Working on Safari 15.1 ✅ 
8. Unit tests passing ✅
9.  Feature is compatible with latest personalized sample data module ✅ 
10. Cypress tests failing ❌ (errors are unrelated to task since they are also happening in https://develop.pwa-venia.com)
